pettingzoo/utils/env.py

```diff
@@ -177,7 +173,7 @@ def agent_iter(self, max_iter: int = 2**63) -> AECIterable:
 
     def last(
         self, observe: bool = True
-    ) -> tuple[ObsType | None, float, bool, bool, dict[AgentID, dict[str, Any]]]:
+    ) -> tuple[ObsType | None, float, bool, bool, dict[str, Any]]:
```

Comment:
Also looking at before and after this line looks incorrect, it should use AgentID, but also simpler to just keep the ObsDict type at the top of the file and define it to use AgentID, which you changed ofc. That way if agent id gets changed again this will not need to be changed as it just references it. And prevents breaking code like the shimmy bug.

Reply:
The reason I changed that line was because we return ` self.infos[agent]` for the info dict, and `self.infos` has the type `dict[AgentID, dict[str, Any]]`.

 That line was producing a type error previously.

Reply:
I'm a little confused why it needs to be dict[str] here though instead of agent ID

Reply:
I guess this has to do with the other PR up right now which talks about infos and the correct syntax for it. What I gather is that parallel envs need to have the dict be indexed by agent ID and have sub dicts with info for each agent, but I guess in AEC environments you just need any dict and it's assumed that it's the info only for that agent on that timestep. So it shouldn't be nested dict like the original code, but I think it should still be dict[AgentID, Any]

Reply:
Well in this case, (`last()`) we're only returning the info for the current agent, so I don't understand why the it should be indexed by AgentID. 

If someone wants to use `int` to index their agents, but still have an info that looks like `{ "extra_info": 1, "extra_info2": 2 }`, (per agent) then that seems fine to me.